dbt/adapters/default/impl.py

```diff
@@ -273,17 +304,51 @@ def expand_target_column_types(cls, profile,
                 cls.alter_column_type(profile, to_schema, to_table,
                                       column_name, new_type, model_name)
 
+    ###
+    # RELATIONS
+    ###
+    @classmethod
+    def list_relations(cls, profile, model_name=None):
+        raise dbt.exceptions.NotImplementedException(
+            '`list_relations` is not implemented for this adapter!')
+
+    @classmethod
+    def get_relation(cls, profile, relations_list=None,
+                     model_name=None, **kwargs):
+        if relations_list is None:
+            relations_list = cls.list_relations(profile, model_name)
```

Comment:
one interesting benefit of grabbing _all_ relations is that the query will be identical for each model. In the future, I think dbt should cache these query results, but in the near-term, Redshift will cache the result for us. Before, we were essentially cache-busting ourselves by filtering on schema and table name!